src/event/cloudevent.ts

```diff
@@ -126,13 +126,13 @@ See: https://github.com/cloudevents/spec/blob/v1.0/spec.md#type-system`);
     Object.freeze(this);
   }
 
-  get data(): unknown {
-    return this.#_data;
+  get data(): T {
+    return this.#_data as T;
   }
 
-  set data(value: unknown) {
+  set data(value: T) {
     if (isBinary(value)) {
```

Comment:
That's a pretty neat feature that I didn't know about. Thanks for pointing it out. I don't understand, though, how it would be an improvement over the existing definition of `isBinary`

```js
export const isBinary = (v: unknown): boolean => v instanceof Uint32Array;
```

Reply:
It allows you to remove cast on line 135. 

```typescript
if (isBinary(value)) {
  this.data_base64 = value;
}
```

Note this would also have passed the type checker:

```typescript
if (value instanceof Uint32Array) {
  this.data_base64 = value;
}
```

Because Typescript understands what `instanceof` means: https://www.typescriptlang.org/docs/handbook/advanced-types.html#instanceof-type-guards

The current `isBinary` function is losing type information because it only returns a `boolean`.

Reply:
Ahhh - that is much nicer!